flang/lib/Lower/OpenMP/ClauseProcessor.h

```diff
@@ -185,7 +184,12 @@ template <typename T>
 bool ClauseProcessor::processMotionClauses(
```

Comment:
I agree and have been thinking about it, as there isn't a whole lot of dissimilarities, most changes to processMapClauses need replicated to processMotionClauses and it would make testing a lot simpler! But it'd also be up to @ergawy I am unsure if he'd like to keep them distinct and has some insights into motion clauses that might make merging the two functions unideal! 

But I can make a follow up PR after the stack has landed to merge them if we are all happy doing so.